File: lachancuoicgoi_flutter/lib/analysis/l3/core/gemini_config.dart
class GeminiConfig {
  const GeminiConfig({
    this.modelName = 'gemini-3.5-flash',
    required this.temperature,
    required this.topK,
    required this.topP,
    this.timeout = const Duration(milliseconds: 25000),
    this.responseMimeType,
  });

  final String modelName;
  final double temperature;
  final int topK;
  final double topP;
  final Duration timeout;
  final String? responseMimeType;

  /// Config cho phân tích L3 (classification với JSON response).
  /// Dùng temperature thấp + topK=1 (greedy) để output ổn định, deterministic.
  static GeminiConfig forAnalysis() {
    return const GeminiConfig(
      modelName: 'gemini-3.5-flash',
      temperature: 0.1,
      topK: 1,
      topP: 1.0,
      timeout: Duration(seconds: 15),
      responseMimeType: 'application/json',
    );
  }

  /// Config cho tóm tắt cuộc gọi (cần độ sáng tạo vừa phải).
  static GeminiConfig forSummarization() {
    return const GeminiConfig(
      modelName: 'gemini-3.5-flash',
      temperature: 0.7,
      topK: 20,
      topP: 0.95,
      timeout: Duration(seconds: 30),
    );
  }
}

File: lachancuocgoi/app/src/main/java/com/example/lachancuocgoi/Analysis/L3/core/GeminiConfig.kt
package com.example.lachancuocgoi.Analysis.L3.core

/**
 * Configuration cho Gemini model.
 * Tách riêng config để dễ dàng tune parameters và testing.
 */
data class GeminiConfig(
    val modelName: String = "gemini-2.5-flash-lite",
    val temperature: Float,
    val topK: Int,
    val topP: Float,
    val timeoutMs: Long = 7000L,
    val responseMimeType: String? = null
) {
    companion object {
        /**
         * Config tối ưu cho phân tích lừa đảo.
         * Cần độ chính xác và consistency cao.
         */
        fun forAnalysis() = GeminiConfig(
            modelName = "gemini-2.5-flash-lite",
            temperature = 0.2f,  // Low variance cho kết quả ổn định
            topK = 20,          // Moderate choices
            topP = 0.9f,        // Balance giữa deterministic và flexibility
            responseMimeType = "application/json" // Yêu cầu trả về JSON structure cho GD2
        )
        
        /**
         * Config tối ưu cho tóm tắt cuộc gọi.
         * Cho phép creativity cao hơn để tạo summaries đa dạng.
         */
        fun forSummarization() = GeminiConfig(
            temperature = 0.7f,  // Higher creativity
            topK = 40,          // More variations
            topP = 0.95f        // High diversity
        )
    }
}
